### PR TITLE
[Feature] [Ready for Code Review] Consolidate expand/collapse use in two locations

### LINKED ISSUE #3689: View-only links - Expand and collapse buttons for component listing are awkward
### Steps
1. Go to [osf.io](http://www.osf.io)
2. Go to one of your projects that has at least two components
3. Go to the project's Sharing page
4. Add a new View-only link that includes all the components

### Expected behavior
I expected that the list of components would be expanded with a double down-arrow icon like in user settings for add-ons.

screen shot 2015-07-20 at 11 32 27 am

### Actual behavior
The expand and collapse buttons are full-size and ungainly. 

screen shot 2015-07-20 at 11 31 27 am
screen shot 2015-07-20 at 11 31 32 am

This is especially apparent when you look at the buttons in the context of the rest of the content under View-only Links.

screen shot 2015-07-20 at 11 36 21 am

If the double down-arrow is not a good solution, then simply having the text "Show 1 more..." and "Collapse" would be better than the full-size buttons.

### PR DESCRIPTION
**Pull request by caneruguz**
## Purpose
Based on issue #3689 this PR replaces the expand/collapse functionality of two locations
- View Only Link table in Sharing page
- Addon active projects list in user settings page

closes #3689 

## Changes
Removes existing expand collapse logic and markup from the two locations and replaces with the osfToggleHeight plugin. 

## Known Side Effects
None

## Screenshots of New Implementation
(see issue #3689 for originals)

screen shot 2015-08-20 at 3 12 25 pm

screen shot 2015-08-20 at 2 35 58 pm

### CODE REVIEW

**Comment by caneruguz** [comment]
After talking with @erinmayhood I should add some notes on this feature for QA:
This fix utilizes an existing expand/collapse plugin to unify two other instances. Now there are three instances where this is used:
- Account Settings: Configure Add-on Accounts: table of projects that have the add-on. 
- Sharing: Add Private Link: table that shows the private link project and subcomponents
- Project Summary: List of contributors. 
If the core plugin toggleHeight is changed in subsequent PR's, all these three locations need to be tested for functionality)

About uniformity: 
This expand/collapse plugin is not designed to replace all instances of expanding. It can be used where  some content needs to be shown to give people there is more of the same. That's why some content in these cases is always visible. The toggle is also always under the cut content so people know it will expand. 

It is NOT intended for locations where the expand/collapse happens in a different location or completely closes/hides certain sections. For instance Panels have their own expand/collapse plugin, treebeard and component widgets have their own functions.  

It also required height to be changeable, so if a component has its own height management this plugin will not work (i.e. treebeard has its own height and toggle management).

File: website/templates/project/contributors.mako [hunk 1]
```diff
@@ -136,14 +136,13 @@
                             </div>
                         </td>
                         <td class="col-sm-4">
-                           <ul class="narrow-list list-overflow" data-bind="foreach: nodesList">
+                           <ul class="private-link-list narrow-list" data-bind="foreach: nodesList, attr: {id: 'private-link-list-' + $index() }">
```
Comment (sloria) [inline website/templates/project/contributors.mako]: @caneruguz Would it be difficult to remove the constraint that osfToggleHeight must be passed elements with IDs? It seems unnecessary.

File: website/static/js/privateLinkTable.js [hunk 2]
```diff
@@ -163,6 +147,8 @@ function ViewModel(url, nodeIsPublic) {
         clipboard(target[0]);
         $tr.find('.remove-private-link').tooltip();
         setupEditable(elm, data);
+        $('.private-link-list').osfToggleHeight({height: 50});
+        $('.toggle-height-gradient').css('left', '0');
```
Comment (sloria) [inline website/static/js/privateLinkTable.js]: Why is this necessary?

**Comment by sloria** [comment]
Pass finished.

File: website/static/js/privateLinkTable.js [hunk 3]
```diff
@@ -5,11 +5,12 @@ var ko = require('knockout');
 var bootbox = require('bootbox');
 var $osf = require('./osfHelpers');
 var clipboard = require('./clipboard');
+require('js/osfToggleHeight');
 
 require('bootstrap-editable');
 
 var ctx = window.contextVars;
-var LINK_CUTOFF = 2;
+//var LINK_CUTOFF = 2;
```
Comment (lyndsysimon) [inline website/static/js/privateLinkTable.js]: Is there a need to leave this line in place? I would much prefer it to be gone.

Comment (caneruguz) [inline website/static/js/privateLinkTable.js]: Oh yes, that should have been taken out. I'll do it now.

**Comment by lyndsysimon** [comment]
First pass complete. :star:
